py/client-ticking/src/pydeephaven_ticking/_core.pyx

```diff
@@ -247,63 +247,59 @@ cdef class ColumnSource:
             self._fill_char_chunk(rows, dest_data, null_flags_ptr)
-            arrow_type = pa.uint16()
         elif element_type_id == ElementTypeId.kInt8:
             dest_data = np.zeros(size, np.int8)
             self._fill_primitive_chunk[int8_t](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.int8()
         elif element_type_id == ElementTypeId.kInt16:
             dest_data = np.zeros(size, np.int16)
             self._fill_primitive_chunk[int16_t](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.int16()
         elif element_type_id == ElementTypeId.kInt32:
             dest_data = np.zeros(size, np.int32)
             self._fill_primitive_chunk[int32_t](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.int32()
         elif element_type_id == ElementTypeId.kInt64:
             dest_data = np.zeros(size, np.int64)
             self._fill_primitive_chunk[int64_t](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.int64()
         elif element_type_id == ElementTypeId.kFloat:
             dest_data = np.zeros(size, np.float32)
             self._fill_primitive_chunk[float](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.float32()
         elif element_type_id == ElementTypeId.kDouble:
             dest_data = np.zeros(size, np.float64)
             self._fill_primitive_chunk[double](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.float64()
         elif element_type_id == ElementTypeId.kBool:
             dest_data = np.zeros(size, np.bool_)
             self._fill_primitive_chunk[bool](rows, dest_data, null_flags_ptr)
-            arrow_type = pa.bool_()
         elif element_type_id == ElementTypeId.kString:
             dest_data = np.zeros(size, object)
             self._fill_string_chunk(rows, dest_data, null_flags_ptr)
-            arrow_type = pa.string()
         elif element_type_id == ElementTypeId.kTimestamp:
             dest_data = np.zeros(size, dtype="datetime64[ns]")
             dest_data_as_int64 = dest_data.view(dtype=np.int64)
             self._fill_timestamp_chunk(rows, dest_data_as_int64, null_flags_ptr)
-            arrow_type = pa.timestamp("ns", tz="UTC")
         elif element_type_id == ElementTypeId.kLocalDate:
             dest_data = np.zeros(size, dtype=np.int64)
             dest_data_as_int64 = dest_data.view(dtype=np.int64)
             self._fill_localdate_chunk(rows, dest_data_as_int64, null_flags_ptr)
-            arrow_type = pa.date64()
         elif element_type_id == ElementTypeId.kLocalTime:
             dest_data = np.zeros(size, dtype=np.int64)
             dest_data_as_int64 = dest_data.view(dtype=np.int64)
             self._fill_localtime_chunk(rows, dest_data_as_int64, null_flags_ptr)
-            arrow_type = pa.time64("ns")
         else:
            raise RuntimeError(f"Unexpected ElementTypeId {<int>element_type_id}")
 
+        arrow_type = _dh_type_to_pa_type(element_type)
```

Comment:
From what I read, we are missing the support of types. See the comments in https://deephaven.atlassian.net/browse/DH-19185. I don't think supporting them is necessary for this feature to ship but want to make a note of it. And I think we can tolerate the discrepancy among the clients wrt types supported, to certain degree.